jcl/src/java.base/share/classes/openj9/internal/tools/attach/target/AttachHandler.java

```diff
@@ -554,7 +554,7 @@ public void run() {
 				if (currentAttachThread.getState() != Thread.State.TERMINATED) {
 					IPC.logMessage("Attach API not terminated"); //$NON-NLS-1$
 				}
-				TargetDirectory.deleteMyDirectory(true); /*[PR Jazz 58094] terminate() cleared out the directory */
+				TargetDirectory.deleteMyDirectory(true, “”); /*[PR Jazz 58094] terminate() cleared out the directory */
```

Comment:
```suggestion
				TargetDirectory.deleteMyDirectory(true, AttachHandler.getVmId()); /*[PR Jazz 58094] terminate() cleared out the directory */
```

Reply:
In this change, `AttachHandler.getVmId()` is going to be invoked regardless of the result of `((!directoryEmpty && !deleteMyFiles()) || (null == targetDirectoryFileObject) || !targetDirectoryFileObject.delete())`, and is subject to `synchronized (accessorMutex)` which is not desirable.
Current code only calls `AttachHandler.getVmId()` when the fast path deletion failed, we should keep that way.

Reply:
I have updae the commit to what I proposed above:
using `Supplier<String> vmIdSupplier` as functional object for vmId, hence we can invoke on demand.

Reply:
AttachAPI initialization occurs at VM early bootstrap time, lambda expressions might cause unexpected class loading behaviour changes.
Please avoid lambda usage here.

Reply:
> AttachAPI initialization occurs at VM early bootstrap time, lambda expressions might cause unexpected class loading behaviour changes.
> Please avoid lambda usage here.

Just for more precise, it's due to performance reason(as mentioned in [here](https://stackoverflow.com/questions/45902215/avoiding-lambda-and-stream-usage-for-a-class-used-at-vm-startup) or unforeseeable behaviour during VM bootstrap per your knowledge?

Reply:
Partially, AttachAPI is not used by many apps which shouldn't be slowdown by its initialization, also there is a functional concern, `initializeAttachAPI` runs asynchronously w/ other VM bootstrap methods and has little co-ordination between them, quite a few error recovering/waiting routines were introduced to mitigate that gap. 
Adding lambda expressions in that stage can complicate its initialization process, and there is no compelling reason to do so.

Reply:
Thanks for your patient answering, Jason.
I will remove lamda on next review.